apps/vaporgui/VActions.h

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include <string>
+
+#include <QWidgetAction>
+
+class VIntSpinBox;
+class VCheckBox;
+class VStringLineEdit;
+class VIntLineEdit;
+class VDoubleLineEdit;
+
```

Comment:
You define a set of VActions where every action, A, creates its corresponding VWidget, W, within a VLineItem, passing its constructor arguments onto W. It then routes W's signals and slots through duplicates in A. This could be simplified by creating a single class VLineAction

```
VLineAction(string title, VWidget *W) : inherits QWidgetAction {
	setDefaultWidget(new VLineItem(title, W));
}
```

which is then used as follows:

```
VCheckbox *check = new VCheckbox;
connect(check, ValueChanged, this, checkChanged);
menu.AddAction(new VLineAction("Title", check));
```

which is almost the same as the original use case:

```
VCheckboxAction *check = new VCheckboxAction("Title");
connect(check, ValueChanged, this, checkChanged);
menu.AddAction(check);
```

while avoiding the need for individual wrappers for every widget.

Reply:
Fair point.